### ISSUE TITLE
Feature: add "requested destination" or other save specs?

### ISSUE THREAD

**Issue by tlambert03** [pymmcore-plus/useq-schema#131]
Should we add something like `requested_destination` to MDASequence?

I don't (immediately) want to add a whole lot of specific saving stuff, but a single field requesting a destination path might be very useful for possible downstream writers: It would provide a directory, base name, and extension.  Writers could then increment that base name as desired (until any further spec added instructions on that sort of thing)

@ianhi ?

**Comment by ianhi** [comment]
Sounds good to me! Maybe also an extra metadata dictionary to allow for downstream uses we can't conceive of now?

Are you imagining these all as required arguments, or optional?

**Comment by tlambert03** [comment]
everything optional.  you should probably always be able to call `MDAEvent()` or `MDASequence()` without args.

**Comment by ianhi** [comment]
ooh right because requested_destination is the name. Rather than something like "save_params" which would be it's own object

**Comment by tlambert03** [comment]
yeah, for now, the writer (over in pymmcore-plus) would have all the parameters.  This is just a bare-minimum sort of start.
I can imagine something like this:

```python
seq = MDASequence(requested_destination='...')

class MDARunner():
    def run(self, events: MDASequence):
        if seq.requested_destination:
            for writer in self.writers:
                writer.set_destination(seq.requested_destination)
```

but... I dunno

**Comment by tlambert03** [comment]
yeah, good question.  I can't see an immediate use case for multiple writers... but I do think that so far, the writer "spec" fits a general category of something that I *can* see multiple of: namely, anything that wants to be notified at the start, end, and each frame of an acquisition, let's call it a handler:

```python
class Handler:
    def sequenceStarted(): ...
    def frameReady(): ...
    def sequenceFinished(): ...
```

so, I'm kinda mentally leaving that possibility open :)
